doc/guides/V8-api-for-node.md

```diff
@@ -0,0 +1,241 @@
+## V8 API usage for Node.js
+
+v8 docs, not particularly useful:
+- https://v8.dev/docs
+
+v8 docs, generated from master:
+- https://v8.paulfryzel.com/docs/master/
+
+v8 code search:
+- https://cs.chromium.org/chromium/src/v8/include/v8.h
+
+Some API info at https://v8.dev/docs/embed
+
+- An isolate is a VM instance with its own heap.
+  Node has one isolate per Worker.
+  Can get from `Local<Object>->GetIsolate()` (but this is slow and discouraged
+  by the V8 people), `Local<Context>->GetIsolate()`, or more typically from
+  `(node::Environment*env)->isolate()` or `FunctionCallbackInfo<Value>
+  args.GetIsolate()`.
+
+- `Local<...>`: A local handle is a pointer to an object. All V8 objects are
+  accessed using handles, they are necessary because of the way the V8 garbage
+  collector works.  Local handles can only be allocated on the stack, in a
+  scope, not with new, and will be deleted as stack unwinds, usually with a
+  scope.
+  - We don't explicitly create HandleScopes very often, because all functions
+    that take `FunctionCallbackInfo` (or similar) already come with one
+    built-in.
+  - Allocating a `Local<>` outside of a scope can lead to difficult to track
+    down memory leaks, so the top-level event loop instantiates a
+    `SealHandleScope`. Basically, the idea is to make sure that the code inside
+    the SealHandleScope always needs to explicitly open a HandleScope if it does
+    use handles, so that they will be deleted when the stack unwinds.
+- Persistent handles last past C++ functions.
+  - PersistentBase::SetWeak trigger a callback from the garbage collector when
+    the only references to an object are from weak persistent handles.
+  - A `v8::Global<>` (alias of a `UniquePersistent<>`) handle relies on C++
+    constructors and destructors to manage the lifetime of the underlying
+    object. We don’t use `UniquePersistent` in Node.js.
```

Comment:
I'd drop the notes about `UniquePersistent`, that's its old name and will probably go away entirely someday.